ext/stack_frames/buffer.c

```diff
@@ -56,11 +61,14 @@ static VALUE buffer_initialize(VALUE self, VALUE size) {
     }
 
     TypedData_Get_Struct(self, buffer_t, &buffer_data_type, buffer);
-    buffer->profile_frames = ALLOC_N(VALUE, capacity);
-    buffer->lines = ALLOC_N(int, capacity);
-    buffer->frames = ALLOC_N(VALUE, capacity);
 
     buffer->capacity = 0;
+    buffer->length = 0;
```

Comment:
Doesn't [`TypedData_Make_Struct` allocate zero'd memory](https://github.com/ruby/ruby/blob/46f6575157d4c2f6bbd5693896e26a65037e5552/include/ruby/internal/core/rtypeddata.h#L458) in the data struct already? Would setting it to 0 here make a difference?

Reply:
Ok, so both lines can go away? I'll simplify the PR.